fineract-provider/src/main/java/org/apache/fineract/accounting/journalentry/service/AccountingProcessorHelper.java

```diff
@@ -889,12 +890,13 @@ private void createDebitJournalEntryForLoan(final Office office, final String cu
         LoanTransaction loanTransaction = null;
         SavingsAccountTransaction savingsAccountTransaction = null;
         ClientTransaction clientTransaction = null;
-        final PaymentDetail paymentDetail = null;
+        PaymentDetail paymentDetail = null;
         final Long shareTransactionId = null;
         String modifiedTransactionId = transactionId;
         if (StringUtils.isNumeric(transactionId)) {
             long id = Long.parseLong(transactionId);
             loanTransaction = this.loanTransactionRepository.findOne(id);
+            paymentDetail = this.loanTransactionRepository.findOne(id).getPaymentDetail();
```

Comment:
unnecessary repository call. You can make use of loanTransaction object to retrieve payment details